src/unittest/test_mesh_compare.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+Minetest
+Copyright (C) 2023 Vitaliy Lobachevskiy
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "test.h"
+
+#include "mesh_compare.h"
+
+class TestMeshCompare : public TestBase {
+public:
+	TestMeshCompare() { TestManager::registerTestModule(this); }
+	const char *getName() override { return "TestMeshCompare"; }
+
+	void runTests(IGameDef *gamedef) override {
+		TEST(testTriangle);
+		TEST(testQuad);
+	}
+
+	void testTriangle() {
```

Comment:
What is the purpose of these tests? What could *possibly* go wrong with triangles?

Reply:
Triangles are fine. Comparing them however...
And this `TestMeshCompare::testTriangle` tests that comparison of triangle-based meshes works as intended. Or at least it should; apparently I forgot to add cases with more than one triangle.

Reply:
So if I understand you correctly, is this to check whether the comparisons within the tests themselves are correct? Basically a test for the unittest?

Reply:
Yes. Anything wrong with that?